source/game/field/obj/ObjectColMgr.cc

```diff
@@ -0,0 +1,49 @@
+#include "ObjectColMgr.hh"
+
+#include "game/system/ResourceManager.hh"
+
+namespace Field {
+
+void *ObjectColMgr::LoadFile(const char *filename) {
+    auto *resMgr = System::ResourceManager::Instance();
+    return resMgr->getFile(filename, nullptr, System::ArchiveId::Course);
+}
+
+/// @addr{0x807C4CE8}
+ObjectColMgr::ObjectColMgr(const char *filename) : m_kclScale(1.0f) {
+    // In the base game, this file is loaded in each GeoObjectKCL and passed into
```

Comment:
This should be a doxygen comment prefixed with `/// @details` rather than have it inside the body of the function.